### ISSUE TITLE
[Android] No message displayed when device in sleep mode 

### ISSUE THREAD

**Issue by bardu** [phonegap/phonegap-plugin-push#1266]
### Expected Behaviour

Notification displayed when the app is in background (user uses a different app) or the device is in sleep mode
### Actual Behaviour

Android: Notification is displayed when user is in a different app but not when device is in sleep mode,
iOS: works as expected
### Reproduce Scenario (including but not limited to)

Using the plugin with topics only
#### Steps to Reproduce
1. Subscribe to a topic
2. send a push notification to this topic
#### Platform and Version

Android 6.0.1
iOS 9.3.5
#### (Android) What device vendor

Samsung S6 Edge
iPhone 6, 4S
#### Cordova CLI version and cordova platform version

```
cordova --version  6.3.1
cordova platform version android  5.2.2 ios 4.2.1
```
#### Plugin version

```
cordova plugin version 1.8.2
```
#### Sample Push Data Payload

```
PushNotification.init({
            android: {
                senderID: config.gcm.senderID,
                sound: "true",
                vibration: "true",
                topics: _.map(androidTopics, 'topic')
            },
            ios: {
                senderID: config.gcm.senderID,
                gcmSandbox: "true",
                alert: "true",
                badge: "true",
                sound: "true",
                vibration: "true",
                clearBadge: "true",
                topics: _.map(iosTopics, 'topic')
            }
        });
```
#### Sample node-gcm

```
// for Android
            var message = new gcm.Message({
                                priority: 'high',
                                delayWhileIdle: true,
                                timeToLive: 3,
                                data: {
                                    url: 'deals',
                                    title: config.gcm.message_title,
                                    body: data.new_val.desc,
                                    sound: 'default',
                                    msgcnt: 1,
                                    contentAvailable: '1'
                                }
                            }),
                sender  = new gcm.Sender(config.gcm.api_key);

                sender.send(message, {topic: '/topics/' + topic.topic.concat('-android')}, function(err, response) {
                    if (err) {
                        console.log('error sending push notification')
                        console.error(err);
                        // TODO: on "NotRegistered" error delete pushId from user db
                    }
                    else {
                        // TODO: here must be taken care of a number of things such as
                        //       see: https://developers.google.com/cloud-messaging/http-server-ref

                        console.log('Push success: ' + JSON.stringify(response));
                    }
                });
            // for iOS 
            message = new gcm.Message({
                                priority: 'high',
                                delayWhileIdle: true,
                                timeToLive: 3,
                                content_available: true,
                                notification: {
                                    url: 'deals',
                                    title: config.gcm.message_title,
                                    body: data.new_val.desc,
                                    sound: 'default',
                                    badge: 1
                                }
                            }),
                sender  = new gcm.Sender(config.gcm.api_key);

                sender.send(message, {topic: '/topics/' + topic.topic.concat('-ios')}, function(err, response) {
                    if (err) {
                        console.log('error sending push notification')
                        console.error(err);
                        // TODO: on "NotRegistered" error delete pushId from user db
                    }
                    else {
                        // TODO: here must be taken care of a number of things such as
                        //       see: https://developers.google.com/cloud-messaging/http-server-ref

                        console.log('Push success: ' + JSON.stringify(response));
                    }
                });
```

**Comment by macdonst** [comment]
@bardu sorry, can you define "sleep mode" for me just so we are on the same page.

**Comment by bardu** [comment]
@macdonst Sorry, if this is miss leading. Shortly press the power button, screen turns black, device is locked but not turned off.

**Comment by adde-wik** [comment]
I've heard similar things from a user of my app, where no notification was received while screen was turned off. I tried today using Android 6.0.1 but was not able to reproduce the "error", my device received the notification in all states (in app, outside app, screen off).

Would be interesting to hear if you resolved your issues.

**Comment by mamutkin** [comment]
I've this problem too. Android 5.1
Anyone solved this problem?

**Comment by adde-wik** [comment]
We pinpointed the issue to Huaweii devices in our case and it can be solved by users following this article,

http://www.forbes.com/sites/bensin/2016/07/04/push-notifications-not-coming-through-to-your-huawei-phone-heres-how-to-fix-it/#16f2146378aa
